src/menu/charprofile.c

```diff
@@ -0,0 +1,403 @@
+/*
+ * This software is licensed under the terms of the MIT License.
+ * See COPYING for further information.
+ * ---
+ * Copyright (c) 2011-2019, Lukas Weber <laochailan@web.de>.
+ * Copyright (c) 2012-2019, Andrei Alexeyev <akari@taisei-project.org>.
+ */
+
+#include "events.h"
+#include "resource/resource.h"
+#include "taisei.h"
+
+#include "charprofile.h"
+#include "mainmenu.h"
+#include "portrait.h"
+#include "common.h"
+#include "progress.h"
+#include "options.h"
+#include "global.h"
+#include "util/glm.h"
+#include "video.h"
+
+#define NUM_PROFILES 11
+#define LOCKED_PROFILE NUM_PROFILES-1
+#define DESCRIPTION_WIDTH (SCREEN_W / 3 + 80)
+
+#define FACES(...) { __VA_ARGS__, NULL }
+
+typedef enum {
+	PROFILE_REIMU,
+	PROFILE_MARISA,
+	PROFILE_YOUMU,
+	PROFILE_CIRNO,
+	PROFILE_HINA,
+	PROFILE_SCUTTLE,
+	PROFILE_WRIGGLE,
+	PROFILE_KURUMI,
+	PROFILE_IKU,
+	PROFILE_ELLY,
+	PROFILE_LOCKED,
+} CharProfiles;
```

Comment:
* You don't need to define `LOCKED_PROFILE` since `PROFILE_LOCKED` exists.
* You can also specify `NUM_PROFILES` implicitly as the last entry in the `CharProfiles` enum.